### ISSUE TITLE
Semantic 02 colors are incorrect in dark mode

### ISSUE THREAD

**Issue by felixfbecker** [sourcegraph/sourcegraph#21601]
When navigating to a settings page on dark theme in redesign, the "Save settings" button looks like this:

![image](https://user-images.githubusercontent.com/10532611/120373696-eeebe300-c318-11eb-8c63-bdac529f63d3.png)

which is the same as in light theme:

![image](https://user-images.githubusercontent.com/10532611/120373872-28bce980-c319-11eb-8c36-75f51228f314.png)

With the redesign off it adapts the color properly:

![image](https://user-images.githubusercontent.com/10532611/120373905-32dee800-c319-11eb-84bc-59f56c5b9e08.png)

![image](https://user-images.githubusercontent.com/10532611/120373947-383c3280-c319-11eb-97f1-ded3cb8dc673.png)

Perhaps a specificity issue that could be solved with `:where()`.

cc @eseliger

**Comment by eseliger** [comment]
Hmm interesting, I didn't touch this 🤔

**Comment by umpox** [comment]
This button is disabled, so I believe this is following the [design](https://www.figma.com/file/NIsN34NH7lPu04olBzddTw/Design-Refresh-Systemization-source-of-truth?node-id=908%3A5827). @sourcegraph/design I wonder if we should consider changing the colors here, the contrast doesn't seem great in dark mode

**Comment by rrhyne** [comment]
Disabled buttons in dark mode should be darker, night lighter. Primary and Secondary follow this pattern, but not danger, success, warning or info. This is caused by flaw in the design system. Note the difference between the 02 Primary and Secondary colors and the 02 Success color:

![image](https://user-images.githubusercontent.com/539268/120576315-6c762880-c3d7-11eb-85f1-ec5c0e88f778.png)

The solution here is to change {danger|success| warning|info}-02 from high brightness to low brightness. This could have some interesting side effects in other places such as alerts, but I'm not sure how widespread the issues are. 

@umpox, my default suggestion here would be to leave the issue unresolved until after the redesign unless you tell us it won't have much of an impact. If that's the case I can provide new {danger|success| warning|info}-02 colors tomorrow.

**Comment by umpox** [comment]
Hey @rrhyne, we've got around to implementing these changes, but I think this might require further discussion. I've split the changes into two PRs to make reviewing the UI diffs easier:

1. Update semantic level 2 colors: https://github.com/sourcegraph/sourcegraph/pull/25030 Check out the [Chromatic diff](https://www.chromatic.com/pullrequest?appId=5f0f381c0e50750022dc6bf7&number=25030) to get real implications of these changes.

2. Update text-muted/disabled: https://github.com/sourcegraph/sourcegraph/pull/25032 Check out the [Chromatic diff](https://www.chromatic.com/pullrequest?appId=5f0f381c0e50750022dc6bf7&number=25032) and the [Percy diff](https://percy.io/Sourcegraph/Sourcegraph/builds/12734511/changed/719532148?browser=chrome&browser_ids=16&subcategories=unreviewed%2Cchanges_requested&viewLayout=overlay&viewMode=original&width=1920&widths=1920).

In summary, I think the first PR is fine (minimal changes) but the second PR has a lot of implications across the application. I especially noticed that `text-disabled` no longer appears to show that an element is disabled in a lot of cases.

Please take a look and let me know your thoughts!
